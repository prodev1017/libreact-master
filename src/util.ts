import * as React from 'react';

export type TComponent<TProps> = React.ComponentClass<TProps> | React.StatelessComponent<TProps>;
export type THoc<P1, P2> = (Comp: TComponent<P1>) => TComponent<P2>

export const noop: (...args) => any = () => {};

export const idx = (obj, accessor) => {
  try {
    return accessor(obj);
  } catch (error) {
    return undefined;
  }
};

export const isClient = typeof window === 'object';

export const on = (obj, ...args) => obj.addEventListener(...args);

export const off = (obj, ...args) => obj.removeEventListener(...args);

export const ns = (name) => `@@libreact/${name}`;

const hasSymbols = typeof Symbol !== 'undefined';

export const sym = (name) => {
  return hasSymbols ? Symbol(name) : ns(name);
};

export const isFn = (fn) => typeof fn === 'function';

export const h = React.createElement;
