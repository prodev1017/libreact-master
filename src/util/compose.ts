const compose = (...functions) => {
  if (!functions.length) {
    return arg => arg;
  }

  if (functions.length === 1) {
    return functions[0];
  }

  return (...args2) => {
    let result = functions[functions.length - 1](...args2);

    for (let i = functions.length - 2; i >= 0; i--) {
      result = functions[i](result);
    }

    return result;
  };
};

export default compose;
