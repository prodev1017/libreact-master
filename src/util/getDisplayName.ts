const getDisplayName = (Comp: any) => {
  const tipof = typeof Comp;

  switch (tipof) {
    case 'string':
      return Comp;
    case 'function':
      return Comp.displayName || Comp.name || 'Unknown';
    case 'object':
      const {type} = Comp;

      return typeof type === 'function' ?
        `<${type.displayName || type.name || String(Comp)}>` :
        `<Unknown>`;
    default:
      return 'Unknown';
  }
}

export default getDisplayName;
