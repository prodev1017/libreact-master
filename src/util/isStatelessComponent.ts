const isStatelessComponent = (Comp) => {
  if (typeof Comp !== 'function') {
    return false;
  }

  return !Comp.prototype || (Comp instanceof Function);
};

export default isStatelessComponent;
