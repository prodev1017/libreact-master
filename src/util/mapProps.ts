import {h} from '../util';

export const mapProps = (mapper) => (Comp) => (props) => h(Comp, mapper(props));
