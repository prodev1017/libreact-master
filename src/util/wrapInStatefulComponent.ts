import * as React from 'react';
import {h} from '../util';

const wrapInStatefulComponent = (Comp) => {
  const Decorated = class Decorated extends React.Component<any, any> {
    render () {
      return h(Comp, this.props);
    }
  };

  return Decorated;
};

export default wrapInStatefulComponent;
