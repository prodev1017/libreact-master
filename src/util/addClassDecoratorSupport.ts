import isStatelessComponent from './isStatelessComponent';
import wrapInStatefulComponent from './wrapInStatefulComponent';

const addClassDecoratorSupport = (Comp) => {
  return isStatelessComponent(Comp) ?
    Comp :
    wrapInStatefulComponent(Comp);
};

export default addClassDecoratorSupport;
