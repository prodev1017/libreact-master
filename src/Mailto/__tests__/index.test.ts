import {mount} from 'enzyme';
import {h} from '../../util';
import {Mailto} from '..';

describe('<Mailto>', () => {
  it('is a component', () => {
    expect(Mailto).toBeInstanceOf(Function);
  });

  it('renders as expected with just e-mail', () => {
    const wrapper = mount(h(Mailto, {
      email: 'foo@bar.baz'
    }));

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders as expected ', () => {
    const wrapper = mount(h(Mailto, {
      email: 'foo@bar.baz',
      subject: 'Hello world',
      cc: ['foo.cc@bar.baz'],
      bcc: ['foo.bcc@bar.baz'],
      body: 'Hello world!'
    }, 'Click me'));

    expect(wrapper.html()).toMatchSnapshot();
  });
});
