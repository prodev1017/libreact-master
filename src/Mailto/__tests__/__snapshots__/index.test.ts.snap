// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mailto> renders as expected  1`] = `"<a href=\\"mailto:foo@bar.baz?subject=Hello%20world&amp;cc=foo.cc%40bar.baz&amp;bcc=foo.bcc%40bar.baz&amp;body=Hello%20world%21\\">Click me</a>"`;

exports[`<Mailto> renders as expected with just e-mail 1`] = `"<a href=\\"mailto:foo@bar.baz?subject=&amp;body=\\"></a>"`;
